src/aiida_quantumespresso/workflows/ph/base.py

```diff
@@ -103,7 +102,23 @@ def get_builder_from_protocol(cls, code, parent_folder=None, protocol=None, over
         if 'settings' in inputs['ph']:
             builder.ph['settings'] = orm.Dict(inputs['ph']['settings'])
         builder.clean_workdir = orm.Bool(inputs['clean_workdir'])
-        builder.ph['qpoints'] = qpoints
+
+        if 'qpoints' in inputs['ph']:
+            qpoints_mesh = inputs['ph'].pop('qpoints')
+            qpoints = orm.KpointsData()
+            qpoints.set_kpoints_mesh(qpoints_mesh)
+            builder.ph['qpoints'] = qpoints
+        else:
+            inputs = {
+                'structure': structure,
+                'distance': orm.Float(inputs['qpoints_distance']),
+                'force_parity': orm.Bool(False),
```

Comment:
Same for this. Would add a default to the protocol, and then get the value from `inputs['force_parity']` allowing a user to override the default value with the `overrides` argument.

Reply:
Then do you think we should also add force_parity and qpoints_distance as required inputs for the workchain?

Reply:
No, I don't think they should be required because it should still be possible to pass explicit `qpoints` directly. I would literally mirror what the `PwBaseWorkChain` does and add them as optional inputs, and make the `qpoints` also optional. Then you add the same validation that either `qpoints` or `qpoints_distance` has to be defined.

Reply:
Yes, that what I mean, thank you! I will do it like that

Reply:
You can add after the input specifications
```python
spec.inputs.validator = validate_inputs
```
and then outside the class something like this:
```python
def validate_inputs(inputs, _):
    """Validate the top level namespace."""
    if 'qpoints_distance' not in inputs and 'qpoints' not in inputs:
            return 'Either `qpoints_distance` or `qpoints` must be given`
```
And then it would be great if you can write a simple test (you can look into the tests folder and look for something similar in the validations of the inputs)